src/core/src/op/convert.cpp

```diff
@@ -2,248 +2,198 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "ngraph/op/convert.hpp"
-
-#include <memory>
-#include <ngraph/validation_util.hpp>
+#include "openvino/op/convert.hpp"
 
+#include "element_visitor.hpp"
 #include "itt.hpp"
-#include "ngraph/op/equal.hpp"
-#include "ngraph/op/select.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/select.hpp"
 #include "openvino/reference/convert.hpp"
+#include "validation_util.hpp"
 
-using namespace std;
-using namespace ngraph;
-
-op::Convert::Convert(const Output<Node>& arg, const element::Type& destination_type)
-    : Op({arg}),
-      m_destination_type(destination_type) {
-    constructor_validate_and_infer_types();
-}
-
-void op::Convert::validate_and_infer_types() {
-    OV_OP_SCOPE(v0_Convert_validate_and_infer_types);
-
-    set_output_type(0, m_destination_type, get_input_partial_shape(0));
-}
-
-bool op::Convert::visit_attributes(AttributeVisitor& visitor) {
-    OV_OP_SCOPE(v0_Convert_visit_attributes);
-    visitor.on_attribute("destination_type", m_destination_type);
-    return true;
-}
-
-shared_ptr<Node> op::Convert::clone_with_new_inputs(const OutputVector& new_args) const {
-    OV_OP_SCOPE(v0_Convert_clone_with_new_inputs);
-    check_new_args_count(this, new_args);
-    return make_shared<Convert>(new_args.at(0), m_destination_type);
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
+namespace ov {
+namespace op {
 namespace convert {
-namespace {
-template <element::Type_t INPUT_ET, element::Type_t OUTPUT_ET>
-bool evaluate(const HostTensorPtr& arg, const HostTensorPtr& out) {
-    out->set_shape(arg->get_shape());
-    size_t element_count = shape_size(out->get_shape());
 
-    if ((INPUT_ET != arg->get_element_type()) || OUTPUT_ET != out->get_element_type()) {
-        return false;
-    }
-    if (((INPUT_ET == element::u1) || (OUTPUT_ET == element::u1)) ||
-        ((INPUT_ET == element::u4) || (OUTPUT_ET == element::u4)) ||
-        ((INPUT_ET == element::i4) || (OUTPUT_ET == element::i4)) ||
-        ((INPUT_ET == element::nf4) || (OUTPUT_ET == element::nf4))) {
-        ov::reference::detail::lp_convert(arg->get_data_ptr<INPUT_ET>(),
-                                          out->get_data_ptr<OUTPUT_ET>(),
-                                          element_count,
-                                          INPUT_ET,
-                                          OUTPUT_ET);
-    } else {
-        ov::reference::convert(arg->get_data_ptr<INPUT_ET>(), out->get_data_ptr<OUTPUT_ET>(), element_count);
-    }
-    return true;
+constexpr bool is_lp_type(const element::Type_t et) {
+    return (et == element::i4) || (et == element::u1) || (et == element::u4) || (et == element::nf4);
 }
 
-#define TYPE_OUT_CASE(a, ...)                                     \
-    case element::Type_t::a: {                                    \
-        OV_OP_SCOPE(OV_PP_CAT3(evaluate_covert_out, _, a));       \
```

Comment:
`OV_OP_SCOPE` not ported, so conditional compilation won't work as previously. @praasz is it possible to support same behaviour?

Reply:
It looks like using `IfTypeOf`` missing this feature. 
I've created the ticket [CVS-124984](https://jira.devtools.intel.com/browse/CVS-124984) to resolve this in next step.
It has to be implemented in generic way for *IfTypeOf* to enable conditional compilation feature.